scipy/stats/_continuous_distns.py

```diff
@@ -7843,17 +7843,38 @@ def _shape_info(self):
         return [ic, i_s]
 
     def _pdf(self, x, c, s):
-        # powerlognorm.pdf(x, c, s) = c / (x*s) * phi(log(x)/s) *
-        #                                         (Phi(-log(x)/s))**(c-1),
-        return (c/(x*s) * _norm_pdf(np.log(x)/s) *
-                pow(_norm_cdf(-np.log(x)/s), c*1.0-1.0))
+
+        def direct_formula(x, c, s):
+            return (c/(x*s) * _norm_pdf(np.log(x)/s) *
+                    pow(_norm_cdf(-np.log(x)/s), c - 1.))
+
+        def logpdf_formula(x, c, s):
+            return np.exp(self._logpdf(x, c, s))
+
+        return _lazywhere((c <= 1) & (s <= 1), (x, c, s, ),
+                          logpdf_formula,
+                          f2=direct_formula)
+
+    def _logpdf(self, x, c, s):
+        return (np.log(c) - np.log(x) - np.log(s) +
+                _norm_logpdf(np.log(x) / s) +
+                _norm_logcdf(-np.log(x) / s) * (c - 1.))
 
     def _cdf(self, x, c, s):
-        return 1.0 - pow(_norm_cdf(-np.log(x)/s), c*1.0)
+        return -sc.expm1(self._logsf(x, c, s))
 
     def _ppf(self, q, c, s):
         return np.exp(-s * _norm_ppf(pow(1.0 - q, 1.0 / c)))
 
+    def _sf(self, x, c, s):
+        return np.exp(self._logsf(x, c, s))
```

Comment:
I always forget whether the distribution infrastructure does this for you. (The new one certainly will.) Could you check? Same with `_pdf` from `_logpdf`, if we decide that the `direct_formula` is not needed.

Reply:
Looking at the source code in [here](https://github.com/scipy/scipy/blob/main/scipy/stats/_distn_infrastructure.py#L1000) it looks like it does not. That would be a great improvement if the new infrastructure took more advantage of the logarithmized methods.